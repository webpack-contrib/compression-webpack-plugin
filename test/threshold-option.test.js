import CompressionPlugin from '../src/index';

import {
  compile,
  getAssetsNameAndSize,
  getCompiler,
  getErrors,
  getWarnings,
  removeCache,
} from './helpers/index';

describe('"threshold" option', () => {
  let compiler;

  beforeEach(() => {
    compiler = getCompiler('./entry.js');

    return removeCache();
  });

  it('matches snapshot for `0` value ({Number})', async () => {
    new CompressionPlugin({
      minRatio: 1,
      threshold: 0,
    }).apply(compiler);

    const stats = await compile(compiler);

    expect(getAssetsNameAndSize(stats)).toMatchSnapshot('assets');
    expect(getWarnings(stats)).toMatchSnapshot('errors');
    expect(getErrors(stats)).toMatchSnapshot('warnings');
  });

  it('matches snapshot for `8192` value ({Number})', async () => {
    new CompressionPlugin({
      minRatio: 1,
      threshold: 8192,
    }).apply(compiler);

    const stats = await compile(compiler);

    expect(getAssetsNameAndSize(stats)).toMatchSnapshot('assets');
    expect(getWarnings(stats)).toMatchSnapshot('errors');
    expect(getErrors(stats)).toMatchSnapshot('warnings');
  });
});
