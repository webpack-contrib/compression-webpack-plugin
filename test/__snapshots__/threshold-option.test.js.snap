// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when applied with \`threshold\` option matches snapshot for \`0\` value ({Number}): assets 1`] = `
Array [
  Array [
    "0.async.dbe59b7ea0e1a63e45d4.js",
    130,
  ],
  Array [
    "0.async.dbe59b7ea0e1a63e45d4.js.gz",
    117,
  ],
  Array [
    "5b1f36bc41ab31f5b801d48ba1d65781.png",
    78117,
  ],
  Array [
    "5b1f36bc41ab31f5b801d48ba1d65781.png.gz",
    73160,
  ],
  Array [
    "96621d3c37d96ad3bf792fcc848d912f.svg",
    672,
  ],
  Array [
    "96621d3c37d96ad3bf792fcc848d912f.svg.gz",
    393,
  ],
  Array [
    "js.8bd7079c9d63d0f6847e.js",
    10232,
  ],
  Array [
    "js.8bd7079c9d63d0f6847e.js.gz",
    2800,
  ],
]
`;

exports[`when applied with \`threshold\` option matches snapshot for \`0\` value ({Number}): errors 1`] = `Array []`;

exports[`when applied with \`threshold\` option matches snapshot for \`0\` value ({Number}): warnings 1`] = `Array []`;

exports[`when applied with \`threshold\` option matches snapshot for \`8192\` value ({Number}): assets 1`] = `
Array [
  Array [
    "0.async.dbe59b7ea0e1a63e45d4.js",
    130,
  ],
  Array [
    "5b1f36bc41ab31f5b801d48ba1d65781.png",
    78117,
  ],
  Array [
    "5b1f36bc41ab31f5b801d48ba1d65781.png.gz",
    73160,
  ],
  Array [
    "96621d3c37d96ad3bf792fcc848d912f.svg",
    672,
  ],
  Array [
    "js.8bd7079c9d63d0f6847e.js",
    10232,
  ],
  Array [
    "js.8bd7079c9d63d0f6847e.js.gz",
    2800,
  ],
]
`;

exports[`when applied with \`threshold\` option matches snapshot for \`8192\` value ({Number}): errors 1`] = `Array []`;

exports[`when applied with \`threshold\` option matches snapshot for \`8192\` value ({Number}): warnings 1`] = `Array []`;
